src/IonTextReader.ts

```diff
@@ -217,20 +217,19 @@ export class TextReader implements Reader {
     }
 
     annotations(): string[] {
-        let ann : string[] = [];
-        for (let str of this._parser.annotations()) {
-            ann.push(str);
-        }
-        for (let i = 0; i < ann.length; i++) {
-            if(ann[i].length > 1 && ann[i][0] === '$') {
-                let tempStr = ann[i].substr(1, ann[i].length);
-                if (+tempStr === +tempStr) {//look up sid, +str === +str is a one line is integer hack
-                    let symbol = this._symtab.getSymbolText(Number(tempStr));
-                    if(symbol === undefined || symbol === null) throw new Error("Unresolvable symbol ID, symboltokens unsupported.");
-                    ann[i] = symbol;
+        let ann: string[] = [];
+        this._parser.annotations().forEach((st) => {
+            let text = st.getText();
+            if (text !== null) {
+                ann.push(text);
+            } else {
+                let symbol = this._symtab.getSymbolText(st.getSid());
+                if (symbol === undefined || symbol === null) {
+                    throw new Error("Unresolvable symbol ID, symboltokens unsupported.");
```

Comment:
I know this isn't your code change, but: I wonder if we should be returning unresolvable symbols as `null` strings instead of throwing an `Error`. That would allow users to at least read the other values returned by `annotations()`  if they're resolvable.

Reply:
ion-hash-js currently expects an exception to be thrown, otherwise it will simply proceed and return an incorrect hash.  `$0` is an edge case, and I'm not inclined to change this behavior until we just go ahead and support symbol tokens.  Additionally, ion-java's `IonReader.getTypeAnnotations()` throws when it encounters `$0` (but `getTypeAnnotationSymbols()` doesn't); that's probably a good precedent to follow.

Reply:
> Additionally, ion-java's IonReader.getTypeAnnotations() throws when it encounters $0 (but getTypeAnnotationSymbols() doesn't); that's probably a good precedent to follow.

Yeah, this was my motivation. In the absence of a `SymbolToken` API, it seemed unfortunate that the only available method would `throw`. However, if `ion-hash-js` depends on this behavior I'm fine with leaving it in place.